### PR TITLE
[CUBRIDMAN-14] [kr] Statement and result cache to improve performance for frequently repeated queries

### PR DESCRIPTION
**Pull request by beyondykk9**
http://jira.cubrid.org/browse/CUBRIDMAN-14

The result cache as known query cache in previous version is refactored in version 11.

The configurable parameter is 3 as below but the they are enabled newly.

QUERY_CACHE_MODE
MAX_QUERY_CACHE_ENTRIES
MAX_QUERY_CACHE_PAGES
The above parameters affect the following sections:

system parameter related
query processing related
csql related (for query cache information display)
etc.

This is Korean version.

### CODE REVIEW

File: ko/admin/config.rst [hunk 1]
```diff
@@ -1775,6 +1779,39 @@ CUBRID 설치 시 생성되는 기본 데이터베이스 환경 설정 파일(**
 
     **max_filter_pred_cache_entries** 는 메모리에 캐시하는 필터링된 인덱스 표현식의 최대 개수를 설정하는 파라미터이다. 필터링된 인덱스 표현식은 컴파일된 상태로 저장되므로, 서버에서 즉시 사용할 수 있다. 캐시에 저장되어 있지 않을 경우, 필터링된 인덱스 표현식을 데이터베이스 스키마에서 가져와서 해석하는 과정이 필요하다.
 
+.. _query-cache-parameters:
+
+Query Cache-Related Parameters
```
Comment (hgryoo) [inline ko/admin/config.rst]: please translate in Korean.

File: ko/admin/config.rst [hunk 2]
```diff
@@ -1775,6 +1779,39 @@ CUBRID 설치 시 생성되는 기본 데이터베이스 환경 설정 파일(**
...
+-----------------------------------
+
+다음은 질의 결과에 대한 캐시 기능과 관련된 파라미터로 각 파라미터의 타입과 설정 가능한 갑의 범위는 다음과 같다.
+
++-------------------------------+--------+----------+----------+----------+
+| Parameter Name                | Type   | Default  | Min      | Max      |
```
Comment (hgryoo) [inline ko/admin/config.rst]: please translate in Korean.

File: ko/sql/tuning.rst [hunk 3]
```diff
@@ -2370,3 +2373,80 @@ SORT-LIMIT 최적화는 **ORDER BY** 절과 LIMIT 절을 명시한 질의에 적
...
+===========
+
+**QUERY_CACHE** 힌트는 반복적으로 실행되는 쿼리의 성능을 향상시키는 데 사용할 수 있으며, 쿼리는 전용 메모리 영역에 캐시되고 그 결과도 별도의 디스크 공간에 캐시된다. 단, 힌트는 SELECT 쿼리에만 적용된다. 그러나 다음과 같은 경우에는 쿼리에 힌트를 적용 할 수 없으며 힌트는 의미가 없어진다.
+
+* 아래와 같이 질의에 시스템 시간 또는 날짜 관련 속성이 포함된 경우
+    예) SELECT SYSDATE, ADDDATE (SYSDATE, INTERVAL -24 HOUR), ADDDATE (SYSDATE, -1);
+* SERIAL 관련 속성이 포함된 경우.
+* 컬럼 경로 관련 속성이 포함된 경우.
+* 메서드가 포함된 경우.
+* 저장 프로 시저 또는 저장 함수가 포함된 경우
```
Comment (hgryoo) [inline ko/sql/tuning.rst]:
```suggestion
* 저장 프로 시저 또는 저장 함수가 포함된 경우.
```

File: ko/sql/tuning.rst [hunk 4]
```diff
@@ -2370,3 +2373,80 @@ SORT-LIMIT 최적화는 **ORDER BY** 절과 LIMIT 절을 명시한 질의에 적
...
         cost:  7 card 0
+
+.. _query-cache :
+
+쿼리 캐시
+===========
+
+**QUERY_CACHE** 힌트는 반복적으로 실행되는 쿼리의 성능을 향상시키는 데 사용할 수 있으며, 쿼리는 전용 메모리 영역에 캐시되고 그 결과도 별도의 디스크 공간에 캐시된다. 단, 힌트는 SELECT 쿼리에만 적용된다. 그러나 다음과 같은 경우에는 쿼리에 힌트를 적용 할 수 없으며 힌트는 의미가 없어진다.
+
+* 아래와 같이 질의에 시스템 시간 또는 날짜 관련 속성이 포함된 경우
```
Comment (hgryoo) [inline ko/sql/tuning.rst]:
```suggestion
* 아래와 같이 질의에 시스템 시간 또는 날짜 관련 속성이 포함된 경우.
```

File: ko/admin/config.rst [hunk 5]
```diff
@@ -1775,6 +1779,39 @@ CUBRID 설치 시 생성되는 기본 데이터베이스 환경 설정 파일(**
...
+
+    다음 예는 최대 500개의 질의를 캐시하는 것을 보여주고 있다. ::
+
+        max_query_cache_entries=500
+
+**query_cache_use_pages**
+
+    **query_cache_use_pages** 는 최대 캐시할 수 있는 결과 페이지에 대한 설정 값이다. 이 파라미터 값이 1이상으로 설정되면 설정된 페이지 만큼의 결과가 캐시된다.
+
+    다음 예는 최대 4000 페이지의 결과를 캐시하는 것을 보여주고 있다 ::
```
Comment (hgryoo) [inline ko/admin/config.rst]:
```suggestion
    다음 예는 최대 4000 페이지의 결과를 캐시하는 것을 보여주고 있다. ::
```

**Review by hgryoo — CHANGES_REQUESTED** [review]

File: ko/admin/config.rst [hunk 6]
```diff
@@ -1775,6 +1779,39 @@ CUBRID 설치 시 생성되는 기본 데이터베이스 환경 설정 파일(**
...
++-------------------------------+--------+----------+----------+----------+
+| max_query_cache_entries       | int    | 200      | 0        | INT_MAX  |
++-------------------------------+--------+----------+----------+----------+
+| query_cache_size_in_pages     | int    | 1,000    | 0        | INT_MAX  |
++-------------------------------+--------+----------+----------+----------+
+
+두 파라미터 값 중 하나가 0인 경우, **QUERY_CACHE** 힌트와 상관 없이 질의 결과는 캐시되지 않는다.
+
+**max_query_cache_entries**
+    **max_query_cache_entries** 는 최대 캐시할 수 있는 질의 개수에 대한 설정 값이다. 이 파라미터 값이 1이상으로 설정되면 설정된 수 만큼의 질의가 캐시된다.
```
Comment (hgryoo) [inline ko/admin/config.rst]: format; one more newline like query_cache_size_in_pages below

File: ko/admin/config.rst [hunk 7]
```diff
@@ -1775,6 +1779,39 @@ CUBRID 설치 시 생성되는 기본 데이터베이스 환경 설정 파일(**
...
 
+.. _query-cache-parameters:
+
+질의 캐쉬 관련 파라미터
+-----------------------------------
+
+다음은 질의 결과에 대한 캐시 기능과 관련된 파라미터로 각 파라미터의 타입과 설정 가능한 갑의 범위는 다음과 같다.
+
++-------------------------------+--------+----------+----------+----------+
+| Parameter Name              | 타입   | 기본값  |  최소값   | 최대값  |
```
Comment (hgryoo) [inline ko/admin/config.rst]: Parameter Name -> 파라미터 이름

File: ko/admin/config.rst [hunk 8]
```diff
@@ -1775,6 +1779,39 @@ CUBRID 설치 시 생성되는 기본 데이터베이스 환경 설정 파일(**
 
     **max_filter_pred_cache_entries** 는 메모리에 캐시하는 필터링된 인덱스 표현식의 최대 개수를 설정하는 파라미터이다. 필터링된 인덱스 표현식은 컴파일된 상태로 저장되므로, 서버에서 즉시 사용할 수 있다. 캐시에 저장되어 있지 않을 경우, 필터링된 인덱스 표현식을 데이터베이스 스키마에서 가져와서 해석하는 과정이 필요하다.
 
+.. _query-cache-parameters:
+
+질의 캐쉬 관련 파라미터
+-----------------------------------
+
+다음은 질의 결과에 대한 캐시 기능과 관련된 파라미터로 각 파라미터의 타입과 설정 가능한 갑의 범위는 다음과 같다.
```
Comment (shparkcubrid) [inline ko/admin/config.rst]:
```suggestion
다음은 질의 결과에 대한 캐시 기능과 관련된 파라미터로 각 파라미터의 타입과 설정 가능한 값의 범위는 다음과 같다.
```

File: ko/sql/tuning.rst [hunk 9]
```diff
@@ -2370,3 +2373,80 @@ SORT-LIMIT 최적화는 **ORDER BY** 절과 LIMIT 절을 명시한 질의에 적
...
+* 아래와 같이 질의에 시스템 시간 또는 날짜 관련 속성이 포함된 경우.
+    예) SELECT SYSDATE, ADDDATE (SYSDATE, INTERVAL -24 HOUR), ADDDATE (SYSDATE, -1);
+* SERIAL 관련 속성이 포함된 경우.
+* 컬럼 경로 관련 속성이 포함된 경우.
+* 메서드가 포함된 경우.
+* 저장 프로 시저 또는 저장 함수가 포함된 경우.
+* dual, _db_attribute 등과 같은 시스템 테이블이 포함된 경우.
+* sys_guid ()와 같은 시스템 함수가 포함된 경우.
+
+힌트가 설정되고 새 SELECT 질의가 처리 될 때 질의가 캐시에서 발견되면 쿼리 캐시를 사용한다. 동일한 데이터베이스에서 동일한 질의 텍스트와 동일한 바인딩 값을 사용하는 경우 질의는 동일한 것으로 간주된다. 캐시 된 질의를 찾을 수없는 경우 질의가 처리된 다음 결과와 함께 캐시가 생성된다. 질의가 캐시에서 발견되면 캐시된 영역에서 결과를 가져온다. CSQL에서는 아래 예제와 같이 COUNT 함수를 사용하여 질의를 반복적으로 실행할 때 개선된 성능을 쉽게 측정 할 수 있다. 첫번째 질의에 대한 결과는 캐시가 되어 있지 않아서 느리지만, 두 번째 질의의 결과는 캐시 된 영역에서 가져오므로 응답 시간이 이전 질의보다 훨씬 빠르다. ::
```
Comment (shparkcubrid) [inline ko/sql/tuning.rst]:
```suggestion
힌트가 설정되고 새 SELECT 질의가 처리될 때 질의가 캐시에서 발견되면 쿼리 캐시를 사용한다. 동일한 데이터베이스에서 동일한 질의 텍스트와 동일한 바인딩 값을 사용하는 경우 질의는 동일한 것으로 간주된다. 캐시 된 질의를 찾을 수 없는 경우 질의가 처리된 다음 결과와 함께 캐시가 생성된다. 질의가 캐시에서 발견되면 캐시된 영역에서 결과를 가져온다. CSQL에서는 아래 예제와 같이 COUNT 함수를 사용하여 질의를 반복적으로 실행할 때 개선된 성능을 쉽게 측정 할 수 있다. 첫번째 질의에 대한 결과는 캐시가 되어 있지 않아서 느리지만, 두 번째 질의의 결과는 캐시된 영역에서 가져오므로 응답 시간이 이전 질의보다 훨씬 빠르다. ::
```

**Review by shparkcubrid — APPROVED** [review]

**Review by hgryoo — APPROVED** [review]

**Pull request merged**